src/components/CourseItem.vue

```diff
@@ -68,17 +62,83 @@
       <v-divider class="lightGray"></v-divider>
     </v-list>
     <v-card-actions class="white">
-      <v-btn text class="font-weight-bold darkBlue--text"> Edit </v-btn>
-      <v-btn text class="font-weight-bold darkerRed--text"> Delete </v-btn>
+      <v-dialog v-model="editDialog" persistent max-width="600px">
+        <template v-slot:activator="{ on, attrs }">
+          <v-btn
+            text
+            class="font-weight-bold darkBlue--text"
+            v-bind="attrs"
+            v-on="on"
+          >
+            Edit
+          </v-btn>
+        </template>
+        <CourseItemEdit
+          @closeCourseDialogEvent="closeEditDialog"
+          @openCourseSnackbarEvent="openSnackBar"
+          :course="course"
+        ></CourseItemEdit>
+      </v-dialog>
+      <v-dialog v-model="deleteDialog" persistent max-width="600px">
+        <template v-slot:activator="{ on, attrs }">
+          <v-btn
+            text
+            class="font-weight-bold darkerRed--text"
+            v-bind="attrs"
+            v-on="on"
+          >
+            Delete
+          </v-btn>
+        </template>
+        <CourseItemDelete
+          @closeCourseDialogEvent="closeDeleteDialog"
+          @openCourseSnackbarEvent="openSnackBar"
+          :course="course"
+        ></CourseItemDelete>
+      </v-dialog>
     </v-card-actions>
   </v-card>
 </template>
 
 <script>
+import CourseItemEdit from "./CourseItemEdit.vue";
+import CourseItemDelete from "./CourseItemDelete.vue";
+
 export default {
   name: "courseItem",
+  data() {
+    return {
+      editDialog: false,
+      deleteDialog: false,
+    };
+  },
+  components: {
+    CourseItemEdit,
+    CourseItemDelete,
+  },
   props: {
     course: Object,
   },
+  methods: {
+    closeEditDialog(val) {
+      this.editDialog = val;
+    },
+    closeDeleteDialog(val) {
+      this.deleteDialog = val;
+    },
+    openSnackBar(val) {
+      this.$emit("openCourseSnackbarEvent", val);
+    },
+    createSemesterString() {
+      let ret = "";
+      for (var i = 0; i < this.course.courseSemester.length; i++) {
+        if (i > 0) {
+          ret += ", ";
+        }
+        ret += this.course.courseSemester[i];
+      }
+      return ret.trim();
+    },
+  },
```

Comment:
Use .join() instead of this entire thing
i.e. 'course.courseSemester.join(", ");'

So just get rid of the function entirely and do the join statement above inside the curly braces.